core/ir/x86/decode.c

```diff
@@ -2579,6 +2580,16 @@ check_is_variable_size(opnd_t op)
 static byte *
 decode_common(dcontext_t *dcontext, byte *pc, byte *orig_pc, instr_t *instr)
 {
+    /*
+     * If we're dealing with decoding from synthetic ISA, we don't care about returning
+     * the pc of the next instruction (?), so we just write the encoding in final_pc and
+     * return it.
+     */
+    if (instr_get_isa_mode(instr) == DR_ISA_SYNTH) {
```

Comment:
Let's have a `XXX i#1684: Add cross-arch support with ISA dispatch in a common location; for now we have to dispatch in each target-arch implementation.` comment or somesuch.

Reply:
Done.